### PR TITLE
[FEAT] Blog List API Endpoint

### PR DESCRIPTION
**Pull request by Pythonian**
This endpoint enables visitors to be able to view a paginated list of all blog posts.

## Description

This pull request introduces an endpoint to fetch and return a paginated list of blog post summaries. Visitors can view a list of blog posts created by super admin, ordered by the created date in descending order. The endpoint supports pagination and ensures soft-deleted blog posts are filtered out for users. 

## Related Issue (Link to issue ticket)

The issue for this pull request was approved [here](https://github.com/hngprojects/hng_boilerplate_nestjs/issues/94)

## Motivation and Context

This change is required to enable visitors to retrieve and view a list of blog posts, with pagination support to manage large sets of blog posts.

## How Has This Been Tested?

The endpoint has been tested. The unit tests are in the file `/tests/v1/test_blog_list.py`.

- [X] Tested successful retrieval of paginated blog posts.
- [X] Tested response when no blog posts are present.
- [X] Simulated internal server errors.
- [X] Tested invalid page or page size parameters.
- [X] Ensured invalid HTTP methods return appropriate status codes.

The tests can be run via `pytest tests/v1/test_blog_list.py`
​
## Affected areas of code

- I added a blog route file at `api/v1/routes/blog.py`
- I added a test file at `tests/v1/test_blog_list.py`
- I added a conftest file at `tests/conftest.py` in other for pytest to be able to locate test directory and files
- I changed `from ...main import app` to `from main import app` in the file `tests/database.py` since pytest understands absolute import better than relative imports
- I added the blog route to the `main.py` file.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
      ​
## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: api/v1/routes/blog.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+from sqlalchemy.exc import SQLAlchemyError
+from api.v1.models.blog import Blog
+from api.db.database import get_db
+import logging
+
+blog = APIRouter(prefix="/blogs", tags=["Blog"])
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
```
Comment (Dev-wonderful) [inline api/v1/routes/blog.py:13]: Implement this properly or remove it

File: api/v1/routes/blog.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+            }
+            results.append(blog_dict)
+
+        return {
+            "count": total_count,
+            "next": next_page,
+            "previous": prev_page,
+            "results": results,
+        }
+
```
Comment (Dev-wonderful) [inline api/v1/routes/blog.py:57]: Use `JsonResponseDict` class in the `utils.json_response` module for your response, for standardizing the response according to the convention

**Review by Dev-wonderful — CHANGES_REQUESTED** [review]

File: api/v1/routes/blog.py (hunk 2 above)
Comment (Pythonian) [reply api/v1/routes/blog.py]: I have made the required changes. Check if it conforms to what you expected.

Comment (Pythonian) [reply api/v1/routes/blog.py]: Let me check the failing tests

Comment (Pythonian) [reply api/v1/routes/blog.py]: JsonResponseDict for standardized API responses has been integrated.

**Pull request closed without merging**